### PR TITLE
handle box errors in graphics and graphics3d

### PR DESCRIPTION
**Pull request by mmatera**
Here we go with a proposal to address #964. The idea is that when an invalid graphics primitive is processed, a pink rectangle is drawn instead. Notice that in WMA no error message is shown in the client. Just the pink background is shown in the image, plus a tooltip in the notebook interface.

### CODE REVIEW

**Comment by rocky** [comment]
```
Graphics3D[{Red, Opacity[0.25], Cuboid[{{0, 0, 0}, {0.5, 0.5, 0.5}}]}]
``` 
is probably a little bit closer. Also it looks like the error message is some sort of legend or alt for the image. 

The closest approximation I can come up with in our limited set of graphics is: 

```
Row[{Graphics3D[{Red, Opacity[0.25], Cuboid[{{0, 0, 0}, {0.5, 0.5, 0.5}}]}], "Cilinder is not a graphics primitive"}]]
```

An canned image would also work.

**Comment by mmatera** [comment]
I think now the behaviour is closer, but requires an slight change in Mathics-Django: https://github.com/Mathics3/mathics-django/pull/198

Also, this makes that the option `Background` works again in `Graphics` and `Graphics3D`

File: mathics/builtin/graphics.py [hunk 1]
```diff
@@ -1087,6 +1094,75 @@ def stylebox_style(style, specs):
                     raise BoxExpressionError
             return new_style
 
+        failed = []
+
+        def build_error_box1(style):
```
Comment (mmatera) [inline mathics/builtin/graphics.py:1099]: If we agree that setting the background is the right thing, these two functions can be safely removed.

Comment (rocky) [reply mathics/builtin/graphics.py:1099]: I agree that setting the background is the right thing.

**Comment by rocky** [comment]
> I think now the behaviour is closer, but requires an slight change in Mathics-Django: [Mathics3/mathics-django#198](https://github.com/Mathics3/mathics-django/pull/198)
> 
> Also, this makes that the option `Background` works again in `Graphics` and `Graphics3D`

This is awesome. Thanks!

File: mathics/builtin/graphics.py [hunk 2]
```diff
@@ -1087,6 +1093,75 @@ def stylebox_style(style, specs):
...
+            error_style = style.klass(
+                style.graphics,
+                edge=ERROR_BOX_EDGE_COLOR,
+                face=ERROR_BOX_FACE_COLOR,
+            )
+            if isinstance(self, GraphicsElements):
+                error_primitive_head = Symbol("PolygonBox")
```
Comment (rocky) [inline mathics/builtin/graphics.py:1105]: Let's  define PolygonBox and Polygon3DBox, in `mathics.core.systemsymbols` and move the assignment of the boxes underneath the definitions of  `ERROR_BOX*COLOR`. Thanks.

File: mathics/builtin/graphics.py [hunk 3]
```diff
@@ -1087,6 +1093,75 @@ def stylebox_style(style, specs):
...
+                error_primitive_expression = Expression(
+                    error_primitive_head,
+                    from_python([(-1, -1), (1, -1), (1, 1), (-1, 1), (-1, -1)]),
+                )
+            else:
+                error_primitive_head = Symbol("Polygon3DBox")
+                error_primitive_expression = Expression(
```
Comment (rocky) [inline mathics/builtin/graphics.py:1112]: See above.

**Comment by rocky** [comment]
With some small changes noted I think this is ready.  It is much improved from the original draft. Thanks. 

After this PR we are in a position now to do the relatively simple task of addressing the other kinds of place where we raise a graphics box error of some sort.

File: mathics/builtin/graphics.py (hunk 2 above)
Comment (mmatera) [reply mathics/builtin/graphics.py:1105]: Since the PR in Mathics-django was merged, this is no longer needed.

**Comment by rocky** [comment]
LGTM. Now that we handle RGB opacity in #968 do we want to add 0.25 opacity to the background to match what you reported before?

**Comment by rocky** [comment]
Background should be documented, but we can do that in a separate PR.

**Comment by mmatera** [comment]
> LGTM. Now that we handle RGB opacity in #968 do we want to add 0.25 opacity to the background to match what you reported before?

This and the documentation for the `Background` property is going to be included in another PR.

**Pull request merged**